NIDAQ adapter: blanking property missing

Hi @nicost -

We are testing the new NIDAQ adapter, it is very nice! We are able to setup our digital output port0, control light switches, and it appears that the sequencing is working based on our oscilloscope. However, there aren't any "blanking" associated properties in the device property browser available when we add the device. It is clear from the digital output that the excitation is not being controlled by the "EXPOSURE OUT" signal from the camera. The [NIDAQ webpage ](https://micro-manager.org/NIDAQ) mentions it, so I'm hoping it is available and we just need to set up our .cfg file differently.

We are using a USB-6341, X-Series. What else can we provide to help troubleshoot?

Thanks!!

Feel free to post the errors, maybe I (or Mark) can help.

FWIW, here is the output when I build this in VS:

```
Build started...
1>------ Build started: Project: MMDevice-SharedRuntime, Configuration: Release x64 ------
1>MMDevice.cpp
1>ModuleInterface.cpp
1>MMDevice-SharedRuntime.vcxproj -> C:\projects\micro-manager\mmCoreAndDevices\build\Release\x64\MMDevice-SharedRuntime.lib
2>------ Build started: Project: NIDAQ, Configuration: Release x64 ------
2>NIDAQ.cpp
2>NIAnalogOutputPort.cpp
2>NIDigitalOutputPort.cpp
2>C:\projects\micro-manager\mmCoreAndDevices\DeviceAdapters\NIDAQ\NIDigitalOutputPort.cpp(135,60): warning C4018: '>=': signed/unsigned mismatch
2>C:\projects\3rdpartypublic\boost-versions\boost_1_77_0\boost\math\common_factor_rt.hpp(14,1): warning C4081: expected 'identifier'; found 'string' (compiling source file NIDAQ.cpp)
2>   Creating library C:\projects\micro-manager\mmCoreAndDevices\build\Release\x64\mmgr_dal_NIDAQ.lib and object C:\projects\micro-manager\mmCoreAndDevices\build\Release\x64\mmgr_dal_NIDAQ.exp
2>Generating code
2>Previous IPDB was built with incompatible compiler, fall back to full compilation.
2>All 1878 functions were compiled because no usable IPDB/IOBJ from previous compilation was found.
2>Finished generating code
2>NIDAQ.vcxproj -> C:\projects\micro-manager\mmCoreAndDevices\build\Release\x64\mmgr_dal_NIDAQ.dll
2>Done building project "NIDAQ.vcxproj".
========== Build: 2 succeeded, 0 failed, 0 up-to-date, 0 skipped ==========
```

I've verified that all of the PATH variables are correct and match what you posted above. I am building via the command line because I have nearly no experience with ant, visual studio, etc... I am using VS2019 for the build and DAQMX v9.2.2.

<details>
<summary>Build log</summary>

```
msbuild micromanager.sln /t:"MMCore;NIDAQ" /p:Configuration=Release /p:Platform=x64
Microsoft (R) Build Engine version 16.11.2+f32259642 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

Building the projects in this solution one at a time. To enable parallel build, please add the "-m" switch.
Build started 6/29/2024 4:03:17 PM.
Project "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\micromanager.sln" on node 1 (MMCore;NIDAQ target(s)).
ValidateSolutionConfiguration:
  Building solution configuration "Release|x64".
Project "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\micromanager.sln" (1) is building "C:\Users\qi2lab\Documents\GitHub\mic
ro-manager\mmCoreAndDevices\MMCore\MMCore.vcxproj" (2) on node 1 (default targets).
Project "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\MMCore\MMCore.vcxproj" (2) is building "C:\Users\qi2lab\Documents\GitHu
b\micro-manager\mmCoreAndDevices\MMDevice\MMDevice-SharedRuntime.vcxproj" (3) on node 1 (default targets).
PrepareForBuild:
  Creating directory "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMDevice-SharedRuntime\".
  Creating directory "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\Release\x64\".
  Creating directory "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMDevice-SharedRuntime\MMD
  evice.B8C95F39.tlog\".
InitializeBuildStatus:
  Creating "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMDevice-SharedRuntime\MMDevice.B8C9
  5F39.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
ClCompile:
  C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\bin\HostX86\x64\CL.exe /c /Zi /nologo /W4 /WX- /diagn
  ostics:column /MP /O2 /Oi /D NDEBUG /D _CRT_SECURE_NO_WARNINGS /D _SCL_SECURE_NO_WARNINGS /D _MBCS /Gm- /EHsc /MD /GS /Gy /fp:precise /Zc:wcha
  r_t /Zc:forScope /Zc:inline /Fo"C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMDevice-Share
  dRuntime\\" /Fd"C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\Release\x64\MMDevice-SharedRuntime.pdb" /external:W4 /Gd
   /TP /wd4127 /wd4290 /wd4828 /FC /errorReport:queue /utf-8 Debayer.cpp DeviceUtils.cpp ImgBuffer.cpp MMDevice.cpp ModuleInterface.cpp Property
  .cpp
  Debayer.cpp
  DeviceUtils.cpp
  ImgBuffer.cpp
  MMDevice.cpp
  ModuleInterface.cpp
  Property.cpp
Lib:
  C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\bin\HostX86\x64\Lib.exe /OUT:"C:\Users\qi2lab\Documen
  ts\GitHub\micro-manager\mmCoreAndDevices\build\Release\x64\MMDevice-SharedRuntime.lib" /NOLOGO /MACHINE:X64 "C:\Users\qi2lab\Documents\GitHub\
  micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMDevice-SharedRuntime\Debayer.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMDevice-SharedRuntime\DeviceUtils.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMDevice-SharedRuntime\ImgBuffer.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMDevice-SharedRuntime\MMDevice.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMDevice-SharedRuntime\ModuleInterface.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMDevice-SharedRuntime\Property.obj"
  MMDevice-SharedRuntime.vcxproj -> C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\Release\x64\MMDevice-SharedRuntime.lib
FinalizeBuildStatus:
  Deleting file "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMDevice-SharedRuntime\MMDevice
  .B8C95F39.tlog\unsuccessfulbuild".
  Touching "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMDevice-SharedRuntime\MMDevice.B8C9
  5F39.tlog\MMDevice-SharedRuntime.lastbuildstate".
Done Building Project "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\MMDevice\MMDevice-SharedRuntime.vcxproj" (default targets
).

PrepareForBuild:
  Creating directory "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\".
  Creating directory "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\MMCore.tlog\".
InitializeBuildStatus:
  Creating "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\MMCore.tlog\unsuccessfulbuild
  " because "AlwaysCreate" was specified.
ClCompile:
  C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\bin\HostX86\x64\CL.exe /c /Zi /nologo /W4 /WX- /diagn
  ostics:column /MP /O2 /D MMDEVICE_CLIENT_BUILD /D NOMINMAX /D NDEBUG /D _CRT_SECURE_NO_WARNINGS /D _SCL_SECURE_NO_WARNINGS /D _MBCS /Gm- /EHsc
   /MD /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /GR /Fo"C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\interme
  diates\Release\x64\MMCore\\" /Fd"C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\Release\x64\MMCore.pdb" /external:W4 /G
  d /TP /wd4127 /wd4290 /wd4828 /FC /errorReport:queue /utf-8 CircularBuffer.cpp Configuration.cpp CoreCallback.cpp CoreFeatures.cpp CorePropert
  y.cpp DeviceManager.cpp Devices\AutoFocusInstance.cpp Devices\CameraInstance.cpp Devices\DeviceInstance.cpp Devices\GalvoInstance.cpp Devices\
  HubInstance.cpp Devices\ImageProcessorInstance.cpp Devices\MagnifierInstance.cpp Devices\SerialInstance.cpp Devices\ShutterInstance.cpp Device
  s\SignalIOInstance.cpp Devices\SLMInstance.cpp Devices\StageInstance.cpp Devices\StateInstance.cpp Devices\XYStageInstance.cpp Error.cpp Frame
  Buffer.cpp LibraryInfo\LibraryPathsWindows.cpp LoadableModules\LoadedDeviceAdapter.cpp LoadableModules\LoadedModule.cpp LoadableModules\Loaded
  ModuleImpl.cpp LoadableModules\LoadedModuleImplWindows.cpp Logging\Metadata.cpp LogManager.cpp MMCore.cpp PluginManager.cpp Semaphore.cpp Task
  .cpp TaskSet.cpp TaskSet_CopyMemory.cpp ThreadPool.cpp
  CircularBuffer.cpp
  Configuration.cpp
  CoreCallback.cpp
  CoreFeatures.cpp
  CoreProperty.cpp
  DeviceManager.cpp
  AutoFocusInstance.cpp
  CameraInstance.cpp
  DeviceInstance.cpp
  GalvoInstance.cpp
  HubInstance.cpp
  ImageProcessorInstance.cpp
  MagnifierInstance.cpp
  SerialInstance.cpp
  ShutterInstance.cpp
  SignalIOInstance.cpp
  SLMInstance.cpp
  StageInstance.cpp
  StateInstance.cpp
  XYStageInstance.cpp
  Error.cpp
  FrameBuffer.cpp
  LibraryPathsWindows.cpp
  LoadedDeviceAdapter.cpp
  LoadedModule.cpp
  LoadedModuleImpl.cpp
  LoadedModuleImplWindows.cpp
  Metadata.cpp
  LogManager.cpp
  MMCore.cpp
  PluginManager.cpp
  Semaphore.cpp
  Task.cpp
  TaskSet.cpp
  TaskSet_CopyMemory.cpp
  ThreadPool.cpp
Lib:
  C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\bin\HostX86\x64\Lib.exe /OUT:"C:\Users\qi2lab\Documen
  ts\GitHub\micro-manager\mmCoreAndDevices\build\Release\x64\MMCore.lib" /NOLOGO /MACHINE:X64 "C:\Users\qi2lab\Documents\GitHub\micro-manager\mm
  CoreAndDevices\build\intermediates\Release\x64\MMCore\CircularBuffer.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\Configuration.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\CoreCallback.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\CoreFeatures.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\CoreProperty.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\DeviceManager.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\AutoFocusInstance.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\CameraInstance.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\DeviceInstance.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\GalvoInstance.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\HubInstance.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\ImageProcessorInstance.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\MagnifierInstance.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\SerialInstance.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\ShutterInstance.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\SignalIOInstance.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\SLMInstance.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\StageInstance.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\StateInstance.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\XYStageInstance.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\Error.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\FrameBuffer.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\LibraryPathsWindows.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\LoadedDeviceAdapter.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\LoadedModule.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\LoadedModuleImpl.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\LoadedModuleImplWindows.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\Metadata.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\LogManager.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\MMCore.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\PluginManager.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\Semaphore.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\Task.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\TaskSet.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\TaskSet_CopyMemory.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\ThreadPool.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\Release\x64\MMDevice-SharedRuntime.lib"
  MMCore.vcxproj -> C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\Release\x64\MMCore.lib
FinalizeBuildStatus:
  Deleting file "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\MMCore.tlog\unsuccessful
  build".
  Touching "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\MMCore\MMCore.tlog\MMCore.lastbuilds
  tate".
Done Building Project "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\MMCore\MMCore.vcxproj" (default targets).

Project "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\micromanager.sln" (1) is building "C:\Users\qi2lab\Documents\GitHub\mic
ro-manager\mmCoreAndDevices\DeviceAdapters\NIDAQ\NIDAQ.vcxproj" (4) on node 1 (default targets).
PrepareForBuild:
  Creating directory "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\NIDAQ\".
  Creating directory "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\NIDAQ\NIDAQ.tlog\".
InitializeBuildStatus:
  Creating "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\NIDAQ\NIDAQ.tlog\unsuccessfulbuild"
  because "AlwaysCreate" was specified.
ClCompile:
  C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\bin\HostX86\x64\CL.exe /c /I"C:\Users\qi2lab\Document
  s\GitHub\micro-manager\mmCoreAndDevices\buildscripts\VisualStudio\\..\..\..\..\3rdparty\NationalInstruments\DAQmx_9.2\include" /I"C:\Users\qi2
  lab\Documents\GitHub\micro-manager\mmCoreAndDevices\buildscripts\VisualStudio\\..\..\MMDevice" /I"C:\Users\qi2lab\Documents\GitHub\micro-manag
  er\mmCoreAndDevices\buildscripts\VisualStudio\\..\..\..\..\3rdpartypublic\boost-versions\boost_1_77_0" /Zi /nologo /W4 /WX- /diagnostics:colum
  n /MP /O2 /Oi /GL /D WIN32 /D NDEBUG /D _WINDOWS /D _USRDLL /D _CRT_SECURE_NO_WARNINGS /D _SCL_SECURE_NO_WARNINGS /D _WINDLL /D _UNICODE /D UN
  ICODE /Gm- /EHsc /MD /GS /Gy /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevic
  es\build\intermediates\Release\x64\NIDAQ\\" /Fd"C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x6
  4\NIDAQ\vc142.pdb" /external:W4 /Gd /TP /wd4127 /wd4290 /wd4828 /FC /errorReport:queue /utf-8 NIDAQ.cpp NIAnalogOutputPort.cpp NIDigitalOutput
  Port.cpp
  NIDAQ.cpp
  NIAnalogOutputPort.cpp
  NIDigitalOutputPort.cpp
C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\DeviceAdapters\NIDAQ\NIDigitalOutputPort.cpp(135,60): warning C4018: '>=': signe
d/unsigned mismatch [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\DeviceAdapters\NIDAQ\NIDAQ.vcxproj]
C:\Users\qi2lab\Documents\GitHub\3rdpartypublic\boost-versions\boost_1_77_0\boost\math\common_factor_rt.hpp(14,1): warning C4081: expected 'iden
tifier'; found 'string' (compiling source file NIDAQ.cpp) [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\DeviceAdapters\NIDAQ\
NIDAQ.vcxproj]
Link:
  C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\bin\HostX86\x64\link.exe /ERRORREPORT:QUEUE /OUT:"C:\
  Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\Release\x64\mmgr_dal_NIDAQ.dll" /INCREMENTAL:NO /NOLOGO /LIBPATH:"C:\Users\
  qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\buildscripts\VisualStudio\\..\..\..\..\3rdparty\NationalInstruments\DAQmx_9.2\lib64\msv
  c" /LIBPATH:"C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\buildscripts\VisualStudio\\..\..\..\..\3rdpartypublic\boost-versi
  ons\boost_1_77_0-lib-x64" nidaqmx.lib kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.
  lib uuid.lib odbc32.lib odbccp32.lib /MANIFEST /MANIFESTUAC:"level='asInvoker' uiAccess='false'" /manifest:embed /DEBUG /PDB:"C:\Users\qi2lab\
  Documents\GitHub\micro-manager\mmCoreAndDevices\build\Release\x64\mmgr_dal_NIDAQ.pdb" /SUBSYSTEM:WINDOWS /OPT:REF /OPT:ICF /LTCG:incremental /
  LTCGOUT:"C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\NIDAQ\mmgr_dal_NIDAQ.iobj" /TLBID:1 /
  DYNAMICBASE /NXCOMPAT /IMPLIB:"C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\Release\x64\mmgr_dal_NIDAQ.lib" /MACHINE:
  X64 /DLL "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\NIDAQ\NIDAQ.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\NIDAQ\NIAnalogOutputPort.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\intermediates\Release\x64\NIDAQ\NIDigitalOutputPort.obj"
  "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\Release\x64\MMDevice-SharedRuntime.lib"
     Creating library C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\Release\x64\mmgr_dal_NIDAQ.lib and object C:\Users\q
  i2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\Release\x64\mmgr_dal_NIDAQ.exp
NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxGetSysDevNames [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\Devic
eAdapters\NIDAQ\NIDAQ.vcxproj]
NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxWriteDigitalU8 [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\Devic
eAdapters\NIDAQ\NIDAQ.vcxproj]
NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxWriteAnalogF64 [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\Devic
eAdapters\NIDAQ\NIDAQ.vcxproj]
NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxExportSignal [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\DeviceA
dapters\NIDAQ\NIDAQ.vcxproj]
NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxReadDigitalLines [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\Dev
iceAdapters\NIDAQ\NIDAQ.vcxproj]
NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxClearTask [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\DeviceAdap
ters\NIDAQ\NIDAQ.vcxproj]
NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxCfgChangeDetectionTiming [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDev
ices\DeviceAdapters\NIDAQ\NIDAQ.vcxproj]
NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxStartTask [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\DeviceAdap
ters\NIDAQ\NIDAQ.vcxproj]
NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxCreateDOChan [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\DeviceA
dapters\NIDAQ\NIDAQ.vcxproj]
NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxCreateTask [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\DeviceAda
pters\NIDAQ\NIDAQ.vcxproj]
NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxCreateAOVoltageChan [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\
DeviceAdapters\NIDAQ\NIDAQ.vcxproj]
NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxGetDevAOPhysicalChans [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevice
s\DeviceAdapters\NIDAQ\NIDAQ.vcxproj]
NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxGetDevTerminals [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\Devi
ceAdapters\NIDAQ\NIDAQ.vcxproj]
NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxWriteDigitalU16 [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\Devi
ceAdapters\NIDAQ\NIDAQ.vcxproj]
NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxCfgSampClkTiming [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\Dev
iceAdapters\NIDAQ\NIDAQ.vcxproj]
NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxWriteDigitalU32 [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\Devi
ceAdapters\NIDAQ\NIDAQ.vcxproj]
NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxGetExtendedErrorInfo [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices
\DeviceAdapters\NIDAQ\NIDAQ.vcxproj]
NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxGetErrorString [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\Devic
eAdapters\NIDAQ\NIDAQ.vcxproj]
NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxGetDevDOPorts [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\Device
Adapters\NIDAQ\NIDAQ.vcxproj]
NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxCreateDIChan [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\DeviceA
dapters\NIDAQ\NIDAQ.vcxproj]
NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxGetDevAOVoltageRngs [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\
DeviceAdapters\NIDAQ\NIDAQ.vcxproj]
NIDigitalOutputPort.obj : error LNK2001: unresolved external symbol DAQmxGetPhysicalChanDOPortWidth [C:\Users\qi2lab\Documents\GitHub\micro-mana
ger\mmCoreAndDevices\DeviceAdapters\NIDAQ\NIDAQ.vcxproj]
C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\Release\x64\mmgr_dal_NIDAQ.dll : fatal error LNK1120: 22 unresolved extern
als [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\DeviceAdapters\NIDAQ\NIDAQ.vcxproj]
Done Building Project "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\DeviceAdapters\NIDAQ\NIDAQ.vcxproj" (default targets) --
FAILED.

Done Building Project "C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\micromanager.sln" (MMCore;NIDAQ target(s)) -- FAILED.


Build FAILED.

"C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\micromanager.sln" (MMCore;NIDAQ target) (1) ->
"C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\DeviceAdapters\NIDAQ\NIDAQ.vcxproj" (default target) (4) ->
(ClCompile target) ->
  C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\DeviceAdapters\NIDAQ\NIDigitalOutputPort.cpp(135,60): warning C4018: '>=': sig
ned/unsigned mismatch [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\DeviceAdapters\NIDAQ\NIDAQ.vcxproj]
  C:\Users\qi2lab\Documents\GitHub\3rdpartypublic\boost-versions\boost_1_77_0\boost\math\common_factor_rt.hpp(14,1): warning C4081: expected 'id
entifier'; found 'string' (compiling source file NIDAQ.cpp) [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\DeviceAdapters\NIDA
Q\NIDAQ.vcxproj]


"C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\micromanager.sln" (MMCore;NIDAQ target) (1) ->
"C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\DeviceAdapters\NIDAQ\NIDAQ.vcxproj" (default target) (4) ->
(Link target) ->
  NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxGetSysDevNames [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\Dev
iceAdapters\NIDAQ\NIDAQ.vcxproj]
  NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxWriteDigitalU8 [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\Dev
iceAdapters\NIDAQ\NIDAQ.vcxproj]
  NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxWriteAnalogF64 [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\Dev
iceAdapters\NIDAQ\NIDAQ.vcxproj]
  NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxExportSignal [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\Devic
eAdapters\NIDAQ\NIDAQ.vcxproj]
  NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxReadDigitalLines [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\D
eviceAdapters\NIDAQ\NIDAQ.vcxproj]
  NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxClearTask [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\DeviceAd
apters\NIDAQ\NIDAQ.vcxproj]
  NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxCfgChangeDetectionTiming [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndD
evices\DeviceAdapters\NIDAQ\NIDAQ.vcxproj]
  NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxStartTask [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\DeviceAd
apters\NIDAQ\NIDAQ.vcxproj]
  NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxCreateDOChan [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\Devic
eAdapters\NIDAQ\NIDAQ.vcxproj]
  NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxCreateTask [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\DeviceA
dapters\NIDAQ\NIDAQ.vcxproj]
  NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxCreateAOVoltageChan [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevice
s\DeviceAdapters\NIDAQ\NIDAQ.vcxproj]
  NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxGetDevAOPhysicalChans [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevi
ces\DeviceAdapters\NIDAQ\NIDAQ.vcxproj]
  NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxGetDevTerminals [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\De
viceAdapters\NIDAQ\NIDAQ.vcxproj]
  NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxWriteDigitalU16 [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\De
viceAdapters\NIDAQ\NIDAQ.vcxproj]
  NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxCfgSampClkTiming [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\D
eviceAdapters\NIDAQ\NIDAQ.vcxproj]
  NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxWriteDigitalU32 [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\De
viceAdapters\NIDAQ\NIDAQ.vcxproj]
  NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxGetExtendedErrorInfo [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevic
es\DeviceAdapters\NIDAQ\NIDAQ.vcxproj]
  NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxGetErrorString [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\Dev
iceAdapters\NIDAQ\NIDAQ.vcxproj]
  NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxGetDevDOPorts [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\Devi
ceAdapters\NIDAQ\NIDAQ.vcxproj]
  NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxCreateDIChan [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\Devic
eAdapters\NIDAQ\NIDAQ.vcxproj]
  NIDAQ.obj : error LNK2001: unresolved external symbol DAQmxGetDevAOVoltageRngs [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevice
s\DeviceAdapters\NIDAQ\NIDAQ.vcxproj]
  NIDigitalOutputPort.obj : error LNK2001: unresolved external symbol DAQmxGetPhysicalChanDOPortWidth [C:\Users\qi2lab\Documents\GitHub\micro-ma
nager\mmCoreAndDevices\DeviceAdapters\NIDAQ\NIDAQ.vcxproj]
  C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\build\Release\x64\mmgr_dal_NIDAQ.dll : fatal error LNK1120: 22 unresolved exte
rnals [C:\Users\qi2lab\Documents\GitHub\micro-manager\mmCoreAndDevices\DeviceAdapters\NIDAQ\NIDAQ.vcxproj]

    2 Warning(s)
    23 Error(s)

Time Elapsed 00:00:09.11
```

--- a/DeviceAdapters/NIDAQ/NIDigitalOutputPort.cpp
+++ b/DeviceAdapters/NIDAQ/NIDigitalOutputPort.cpp
@@ -98,6 +98,16 @@ int DigitalOutputPort::Initialize()
       supportsBlankingAndSequencing_ = true;
    GetHub()->StopDOBlankingAndSequence(portWidth_);
 
+   // Some cards lie about their portwidth, if blanking does not work, try 32 bits
+   if (!supportsBlankingAndSequencing_)
+   {
+      uInt32 oldPortWidth = portWidth_;
+      portWidth_ = 32;
+      if (GetHub()->StartDOBlankingAndOrSequence(tmpNiPort, portWidth_, true, false, 0, false, tmpTriggerTerminal) == DEVICE_OK)
+         supportsBlankingAndSequencing_ = true;
+      GetHub()->StopDOBlankingAndSequence(portWidth_);
+   }
+
    CPropertyAction* pAct;
    if (supportsBlankingAndSequencing_)
    {

